pgmax/factors/logical.py

```diff
@@ -197,14 +151,94 @@ class ORFactor(LogicalFactor):
     F(p1, p2, ..., pn, c) = -inf o.w.
     """
 
-    pass
+    def compile_wiring(
+        self, vars_to_starts: Mapping[nodes.Variable, int]
+    ) -> LogicalWiring:
+        """Compile LogicalWiring for the ORFactor
+
+        Args:
+            vars_to_starts: A dictionary that maps variables to their global starting indices
+                For an n-state variable, a global start index of m means the global indices
+                of its n variable states are m, m + 1, ..., m + n - 1
+
+        Returns:
+             LogicalWiring for the ORFactor
+        """
+        var_states_for_edges = np.concatenate(
+            [
+                np.arange(variable.num_states) + vars_to_starts[variable]
+                for variable in self.variables
+            ]
+        )
+        num_parents = len(self.variables) - 1
+        parents_edge_states = np.vstack(
+            [
+                np.zeros(num_parents, dtype=int),
+                np.arange(0, 2 * num_parents, 2, dtype=int),
+            ],
+        ).T
+        child_edge_state = np.array([2 * num_parents], dtype=int)
+        return LogicalWiring(
+            edges_num_states=self.edges_num_states,
+            var_states_for_edges=var_states_for_edges,
+            parents_edge_states=parents_edge_states,
+            children_edge_states=child_edge_state,
+            edge_states_offset=1,
```

Comment:
Should we make edge_states_offset an attribute of the class?

Reply:
Good idea. Changed.